### ISSUE TITLE
Notify User of Important Events

### ISSUE THREAD

**Issue by james-d-elliott** [authelia/authelia#4382]
### Description

Email users when important events occur. This will require the following:

1. [ ] A new email template specific for events.

### Use Case

Examples:

1. [ ] 2FA credential added.
2. [ ] 2FA credential removed.
3. [ ] Failed login.
    - This will require some gatekeeping to prevent email spam.
    - Will likely be a future endeavor.
4. [ ] Login from new IP.
    - Requires logging which remote IP's users have logged in from. 
    - Will likely be a future endeavor.
5. [ ] Login from new device.
    - Requires storing an opaque cryptographically secure value in localStorage.
    - Will likely be a future endeavor.

**Comment by smkent** [comment]
I'd also be interested in email notifications on OpenID Connect client consent authorization.

Making notifications configurable might also be nice but could certainly be a future goal.

**Comment by james-d-elliott** [comment]
Yeah I've added this to the list. Think it's an opt-in and we can add it later like the rest.

**Comment by nightah** [comment]
The opt-in nature kind of goes against the principle of security through notification.

What's your thoughts around how the opt-in would work?

I think long term if a user can see all their OIDC linked apps/consents then opt-in would make sense though.

**Comment by james-d-elliott** [comment]
No, the meat of this has been done we'll defer the remaining parts for 4.39.0. We're just waiting for toolchain related elements (go wasn't up to date in alpine until today). Once we've confirmed all tooling is building with go 1.22.1 we're going to release.

**Comment by cerede2000** [comment]
I see this is milestoned for v4.40.0, so commenting while the design is still open.

The list here is framed around end users ("Email users when important events occur"), which makes sense for the 2FA credential items. I'd like to raise an axis the issue doesn't currently cover: the **operator-facing** side of the "Failed login" item. It's a different audience with different constraints, and I think it's worth being explicit about in the design rather than discovering it later.

Authelia already has a complete ban system — `modes: ['user','ip']`, persistence, and CLI management via `authelia storage bans`. The need for operators to act on bans is therefore already acknowledged. What's missing is any way to *learn* that a ban happened without polling for it.

Today the options are:

1. **Log parsing** — undocumented format, breaks silently on upgrade, and requires filtering benign noise (e.g. the recurring `path=/ remote_ip=<proxy>` 408 timeouts logged at `level=error` on any reverse-proxy setup).
2. **Prometheus metrics** — `banned="true"` is exposed, but as an aggregate counter. It tells you *that* something happened, never *who*. That's correct design for metrics — IPs as labels would cause cardinality explosion — so it can never carry the actionable detail.
3. **Polling `authelia storage bans ip list`** — works, and it's what I ended up doing, but there's no machine-readable output (`--help` shows only `-h`), so it means diffing plain text on a timer.
4. **Reading the database directly** — internal schema, no stability guarantee.

None of these is something I'd want to depend on long-term for a security signal.

On the gatekeeping concern noted against "Failed login": it's real for per-attempt user email, but much smaller for an operator channel triggered on **ban creation** rather than on each attempt. Regulation already provides the debouncing — a ban is by definition a rate-limited event. That might make the operator variant cheaper to ship than the user-facing one.

A minimal `ban.created` payload carrying mode, target, reason and expiry would cover it, and could ride on whatever channel #7695 lands on.

Two questions, if you have visibility:

- Is v4.40.0 expected to cover the remaining unchecked items here, or only a subset? The three checked items suggest the notification plumbing already exists, so I'm trying to gauge whether "Failed login" is in scope for that release or still a later endeavour.
- Would an operator-facing channel be considered part of this issue, or should it be tracked separately?
